products/identity/Products/Sign-In Service/Engineering Docs/Authentication Types/Client Auth (User)/auth_flows/api_oauth.md

```diff
@@ -0,0 +1,194 @@
+
+In order to successfully develop against a local instance of Sign in Service, [vets-api](https://github.com/department-of-veterans-affairs/vets-api) must be set up, either natively or through Docker.
+
+`vets-api` localhost performs a real authentication with the CSP, but relies on mocked user data from MPI. It must be configured to look for this mocked data from [vets-api-mockdata](https://github.com/department-of-veterans-affairs/vets-api-mockdata). Make sure you have the latest version of `vets-api-mockdata` (including running `ruby make_table.rb` in the mock data repository to populate the mock data tables) before attempting to authenticate with SiS to prevent missing mocked data errors.
+
+### Client Config
+
+In order to make use of the Sign in Service, clients must first [register a `Client Configuration`](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity/Products/Sign-In%20Service/Engineering%20Docs/configuration/client_config.md).
+
+Below are the ClientConfig attributes that necessary for a successful mobile or API integration with SiS:
+
+- `authentication`: 'api'
+- `anti_csrf`: `false`
+- `access_token_duration`: 30.minutes
+- `refresh_token_duration`: 45.days
+- `pkce`: true
+- `certificates`: nil
+
+## Sequence Diagram
+
+![pkce_oauth_diagram](https://user-images.githubusercontent.com/71290526/158837072-3e777557-e223-4a9c-948b-dd312f2b88cb.png)[Source](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/vsp/teams/Identity/Product%20Documentation/Sign%20In%20Service/Diagrams/PKCE_Oauth.md)
+
+## Technical Diagram
+
+![image](https://user-images.githubusercontent.com/20125855/177562919-43b99aa7-287b-475b-aa2f-da0e00c05a5c.png)
+
+## Flow Diagrams
+
+### Frontend Client
+
+- [Unauthenticated request flow](../flow%20diagrams/pkce/frontend%20driven/unauthenticated-new-session.png)
+- [Authenticated - access_token expired request flow](../flow%20diagrams/pkce/frontend%20driven/authenticated-expired-access-token.png)
+- [Previously Authenticated - access_token valid request flow](../flow%20diagrams/pkce/frontend%20driven/authenticated-valid-access-token.png)
+
+### Backend Client
+
+- [Authenticated - access_token expired request flow](../flow%20diagrams/pkce/backend%20driven/authenticated-expired-access-token.png)
+- [Authenticated - access_token valid request flow](../flow%20diagrams/pkce/backend%20driven/authenticated-valid-access-token.png)
+
+## Sign in Service Public Routes
+
+The Sign in Service routes necessary for an API integration are listed below. Routes that are authenticated require a valid SiS `access_token` passed through Bearer Auth. Refresh token authenticated routes requires a `refresh_token` request parameter.
+
+### GET Routes
+
+#### [Authorize](../endpoints/authorize.md) 
+- initiates a session with SiS and prompts the user to enter credentials
+
+#### [Introspect](../endpoints/introspect.md)
```

Comment:
I think we deleted this route, we should probably remove the docs for it as well